discovery-provider/alembic/trigger_sql/handle_track.sql

```diff
@@ -9,46 +9,47 @@ begin
   insert into aggregate_track (track_id) values (new.track_id) on conflict do nothing;
   insert into aggregate_user (user_id) values (new.owner_id) on conflict do nothing;
 
+  -- typical "track edits" will mark old row is_current = false and insert a new row
+  -- so find the old_row here:
+  select * into old_row from tracks where track_id = new.track_id and is_current = false order by blocknumber desc limit 1;
 
-  -- increment or decrement?
+  -- but there are some places where we do an "in place" update (like update is_available to false)
   if TG_OP = 'UPDATE' then
     old_row := OLD;
+  end if;
 
-    if (old_row.is_delete = false and new.is_delete = true) or (old_row.is_available = true and new.is_available = false) then
+  -- update aggregate_user.track_count
+  if old_row is not null then
+    -- public track was deleted: decrement
+    if new.is_unlisted = false and ((old_row.is_delete = false and new.is_delete = true) or (old_row.is_available = true and new.is_available = false)) then
```

Comment:
should we also account for delta := 1 if someone edits a track to be unlisted -> listed